src/server/client.ts

```diff
@@ -0,0 +1,505 @@
+/// <reference path="session.ts" />
+
+module ts.server {
+
+    export interface SessionClientHost extends LanguageServiceHost {
+        writeMessage(message: string): void;
+    }
+
+    interface CompletionEntry extends CompletionInfo {
+        fileName: string;
+        position: number;
+    }
+
+    interface RenameEntry extends RenameInfo {
+        fileName: string;
+        position: number;
+        locations: RenameLocation[];
+        findInStrings: boolean;

```

Comment:
Why is this information passed back in the entry API?  this is infrmation the client should be sending into the API when calling it.